statistics/estimate.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statistics
+
+import (
+	"math"
+
+	"github.com/cznic/mathutil"
+)
+
+// calculateEstimateNDV calculates the estimate ndv of a sampled data from a multisize with size total.
+// count[i] stores the count of the i-th element.
+// onlyOnceItems is the number of elements that occurred only once.
+func calculateEstimateNDV(h *topNHelper, rowCount uint64) (ndv uint64, ratio uint64) {
+	sampleSize, sampleNDV, onlyOnceItems := h.sampleSize, uint64(len(h.sorted)), h.onlyOnceItems
+	ratio = mathutil.MaxUint64(1, rowCount/sampleSize)
+
+	if onlyOnceItems == sampleSize {
+		// Assume this is a unique column
+		return rowCount, 1
```

Comment:
Why don't we return the real ratio?

Reply:
In this case, we assume that this column is (almost) unique.

Reply:
what does the `ratio` mean actually?

Reply:
Scale ratio, but only in this case, since we assume this column is unique, it should be 1.

Reply:
ok, we'd better use a comment to explain it.